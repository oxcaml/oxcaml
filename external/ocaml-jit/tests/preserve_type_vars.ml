module A : sig 
  type ('foo, 'bar) t

  val get_foo : ('foo, _) t -> 'foo option
end = struct
  type ('foo, 'bar) t =
    | Foo of 'foo
    | Bar of 'bar

  let get_foo = function
    | Foo foo -> Some foo
    | Bar _ -> None
end
;;

A.get_foo
;;
