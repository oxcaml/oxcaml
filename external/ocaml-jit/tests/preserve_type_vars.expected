module A :
  sig type ('foo, 'bar) t val get_foo : ('foo, 'a) t -> 'foo option end
- : ('foo, 'a) A.t -> 'foo option = <fun>
